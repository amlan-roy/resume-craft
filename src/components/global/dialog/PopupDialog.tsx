import React from "react";
import { AlertDialogProps } from "@radix-ui/react-alert-dialog";
import {
  AlertDialog,
  AlertDialogAction,
  AlertDialogCancel,
  AlertDialogContent,
  AlertDialogDescription,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogTitle,
} from "@/components/ui/alert-dialog";

type PopupDialogProps = {
  onCancel?: () => void;
  onConfirm?: () => void;
  cancelText?: string;
  confirmText?: string;
  description: string;
  open?: boolean;
  title: string;
} & AlertDialogProps;

const PopupDialog: React.FC<PopupDialogProps> = ({
  onCancel,
  onConfirm,
  cancelText = "Cancel",
  confirmText = "Continue",
  description,
  title,
  open,
  ...rest
}) => {
  return (
    <AlertDialog open={open} {...rest}>
      <AlertDialogContent>
        <AlertDialogHeader>
          <AlertDialogTitle>{title}</AlertDialogTitle>
          <AlertDialogDescription>{description}</AlertDialogDescription>
        </AlertDialogHeader>
        <AlertDialogFooter>
          <AlertDialogCancel
            onClick={() => {
              onCancel?.();
            }}
            type="button"
          >
            {cancelText}
          </AlertDialogCancel>
          <AlertDialogAction
            onClick={() => {
              onConfirm?.();
            }}
            type="button"
          >
            {confirmText}
          </AlertDialogAction>
        </AlertDialogFooter>
      </AlertDialogContent>
    </AlertDialog>
  );
};
export default PopupDialog;
