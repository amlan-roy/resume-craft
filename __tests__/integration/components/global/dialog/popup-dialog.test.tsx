import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import PopupDialog from "@/components/global/dialog/PopupDialog";

describe("integration:components/global/PopupDialog", () => {
  it("snapshot", () => {
    const tree = render(
      <PopupDialog
        open
        title="Tu Tu hai wahi dil ne jise apna kaha"
        description="Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
      />
    );
    expect(tree.container).toMatchSnapshot();
  });

  it("Passed in text copies render correctly", () => {
    const tree = render(
      <PopupDialog
        title="Tu Tu hai wahi dil ne jise apna kaha"
        description="Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
        cancelText="Ho mil jaayein is tarah"
        confirmText="do leharein jis tarah"
        open
      />
    );
    expect(
      screen.getByText("Tu Tu hai wahi dil ne jise apna kaha")
    ).toBeInTheDocument();
    expect(
      screen.getByText("Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza")
    ).toBeInTheDocument();
    expect(screen.getByText("Ho mil jaayein is tarah")).toBeInTheDocument();
    expect(screen.getByText("do leharein jis tarah")).toBeInTheDocument();

    expect(tree.container).toMatchSnapshot();
  });

  it("Does not render when open prop is not provided", () => {
    const tree = render(
      <PopupDialog
        title="Tu Tu hai wahi dil ne jise apna kaha"
        description="Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
        cancelText="Ho mil jaayein is tarah"
        confirmText="do leharein jis tarah"
      />
    );
    expect(
      screen.queryByText("Tu Tu hai wahi dil ne jise apna kaha")
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(
        "Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
      )
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText("Ho mil jaayein is tarah")
    ).not.toBeInTheDocument();
    expect(screen.queryByText("do leharein jis tarah")).not.toBeInTheDocument();

    expect(tree.container).toMatchSnapshot();
  });

  it("Calls the passed onCancel callback on click of the cancel button", async () => {
    const onCancel = jest.fn();
    const user = userEvent.setup();
    render(
      <PopupDialog
        open
        onCancel={onCancel}
        title="Tu Tu hai wahi dil ne jise apna kaha"
        description="Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
      />
    );
    await user.click(screen.getByRole("button", { name: /Cancel/i }));
    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  it("Calls the passed onConfirm callback on click of the cancel button", async () => {
    const onConfirm = jest.fn();
    const user = userEvent.setup();
    render(
      <PopupDialog
        open
        onConfirm={onConfirm}
        title="Tu Tu hai wahi dil ne jise apna kaha"
        description="Tu hai jaha mai hu waha ab yeh jeena tere bin hai saza"
      />
    );
    await user.click(screen.getByRole("button", { name: /Continue/i }));
    expect(onConfirm).toHaveBeenCalledTimes(1);
  });
});
