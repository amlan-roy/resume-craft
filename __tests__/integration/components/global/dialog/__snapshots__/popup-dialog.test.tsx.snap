// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/PopupDialog Does not render when open prop is not provided 1`] = `<div />`;

exports[`integration:components/global/PopupDialog Passed in text copies render correctly 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;

exports[`integration:components/global/PopupDialog snapshot 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
